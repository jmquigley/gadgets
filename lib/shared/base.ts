import * as React from 'react';
import {Size} from './index';

const styles = require('./styles.css');

export interface BaseOptions {
	className?: boolean;
	disabled?: boolean;
	visible?: boolean;
}

const defaultBaseOptions: BaseOptions = {
	className: true,
	disabled: true,
	visible: true
};

/**
 * A base class between React and all components in the module.
 */
export abstract class BaseComponent<P, S> extends React.Component<P, S> {

	protected _classes: string[] = [];
	protected _inlineStyle: any = {};    // inline style overrides
	protected _styles: any = {};         // css modules styles

	constructor(props: P, pstyles: any = {}) {
		super(props);
		this._styles = pstyles;
	}

	get classes(): string[] {
		return this._classes;
	}

	set classes(str: string[]) {
		this._classes = str;
	}

	get inlineStyle(): any {
		return this._inlineStyle;
	}

	set inlineStyle(val: any) {
		this._inlineStyle = Object.assign(this._inlineStyle, val);
	}

	get style(): string {
		return this._inlineStyle;
	}

	get styles(): any {
		return this._styles;
	}

	protected buildStyles(props: P, style: any = {}, opts?: BaseOptions): void {
		this._classes = [];
		this._inlineStyle = Object.assign(this._inlineStyle, props['style'], style);

		opts = Object.assign(
			defaultBaseOptions,
			opts
		);

		if (props['className'] !== '' && opts.className) {
			this._classes.push(props['className']);
		}

		if (!props['visible'] && opts.visible) {
			this._classes.push(this.styles.invisible);
		}

		if (props['disabled'] && opts.disabled) {
			this._classes.push(styles.disabled);
			this._classes.push('nohover');
		}
	}

	/**
	 * Takes a sizing parameter and a styles object and searches for the
	 * corresponding style class CSS.  If it is found, then it returns the local
	 * style for that size.
	 * @returns {string} the name of the local style for that size.  If
	 * it is not found, then an empty string is returned.
	 */
	protected getSizeStyle() {

		switch (this.props['size']) {
		case Size.xxsmall:
			if ('xxsmall' in this.styles) {
				return this.styles.xxsmall;
			}
			break;

		case Size.xsmall:
			if ('xsmall' in this.styles) {
				return this.styles.xsmall;
			}
			break;

		case Size.small:
			if ('small' in this.styles) {
				return this.styles.small;
			}
			break;

		case Size.large:
			if ('large' in this.styles) {
				return this.styles.large;
			}
			break;

		case Size.xlarge:
			if ('xlarge' in this.styles) {
				return this.styles.xlarge;
			}
			break;

		case Size.xxlarge:
			if ('xxlarge' in this.styles) {
				return this.styles.xxlarge;
			}
			break;

		case Size.normal:
		case Size.medium:
		default:
			if ('medium' in this.styles) {
				return this.styles.medium;
			}
		}

		return '';
	}
}
