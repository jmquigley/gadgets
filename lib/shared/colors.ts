export type ColorType = string;

export enum ColorScheme {
	c1 = "#004358",
	c2 = "#1f8a70",
	c3 = "#bedb39",
	c4 = "#ffe11a",
	c5 = "#fd7400"
}

export enum Color {
	black = "#000000",
	error = "#d9534f",
	glow = "#51cbee",
	gray = "#666666",
	info = "#5bc0de",
	lightgray = "#cccccc",
	lightsaber = "#2ff923",
	red = "#ff0000",
	silver = "#c0c0c0",
	slategray = "#708090",
	success = "#5cb85c",
	ultralightgray = "#e6e6e6",
	warning = "#f0ad4e",
	white = "#ffffff"
}
