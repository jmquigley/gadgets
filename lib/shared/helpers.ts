"use strict";

const debug = require("debug")("helpers");

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is already in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(
	name: string,
	pkgname: any,
	replace: boolean = false
) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
		debug("Adding global variable %o -> %O", name, (global as any)[name]);
	} else {
		ref = (global as any)[name];
	}

	return ref;
}
