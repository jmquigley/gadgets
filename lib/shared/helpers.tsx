'use strict';

const debugEntry = require('debug');
const pkg = require('../../package.json');

/**
 * A wrapper for the debug function.  It uses the package.json to enable or
 * disable debugging information in the library.  It also turns off
 * coverage checks for this function.
 * @param context {string} the name of the module where this debug request
 * is made.
 * @param args {object[]} the list of arguments passed to the the debug
 * function.
 */
export function debug(context: string, ...args: any[]) {
	/* istanbul ignore if  */
	if (pkg.debug) {
		const debugFn = debugEntry(`Gadgets -> ${context}`);
		debugFn(...args);
	}
}

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is alread in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(name: string, pkgname: any, replace: boolean = false) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
	} else {
		ref = (global as any)[name];
	}

	return ref;
}
