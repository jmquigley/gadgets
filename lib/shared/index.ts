import {
	BaseComponent,
	baseZIndex,
	defaultSize
} from './base';

import {
	getDefaultWrapperProps,
	Wrapper,
	WrapperProps
} from './wrapper';

import {
	BaseProps,
	Color,
	ColorScheme,
	Direction,
	disabled,
	DisabledCSS,
	getDefaultBaseProps,
	invisible,
	InvisibleCSS,
	Justify,
	Location,
	locationStyle,
	SortOrder,
	Styles
} from './props';

import {
	borderStyle,
	boxStyle,
	fontStyle,
	FontStyle,
	lineHeightStyle,
	rectStyle,
	Sizes,
	Sizing,
	Styling
} from './sizing';

import {
	BaseState,
	getDefaultBaseState
} from './state';

import {
	getTheme,
	getThemeList,
	setTheme,
	Theme,
	ThemeProps
} from './themes';

export {
	BaseComponent,
	BaseProps,
	BaseState,
	baseZIndex,
	borderStyle,
	boxStyle,
	Color,
	ColorScheme,
	defaultSize,
	Direction,
	disabled,
	DisabledCSS,
	fontStyle,
	FontStyle,
	getDefaultBaseProps,
	getDefaultBaseState,
	getDefaultWrapperProps,
	getTheme,
	getThemeList,
	invisible,
	InvisibleCSS,
	Justify,
	lineHeightStyle,
	Location,
	locationStyle,
	rectStyle,
	setTheme,
	Sizes,
	Sizing,
	SortOrder,
	Styles,
	Styling,
	Theme,
	ThemeProps,
	Wrapper,
	WrapperProps
};
