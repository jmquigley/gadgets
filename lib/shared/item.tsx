'use strict';

import {cloneDeep} from 'lodash';
import * as React from 'react';
import {BaseProps, getDefaultBaseProps} from './props';
import {Title, TitleProps} from '../title';

const styles = require('./styles.css');

export interface ItemProps extends BaseProps, TitleProps {
	hiddenLeftButton?: boolean;
	hiddenRightButton?: boolean;
	leftButton?: any;
	rightButton?: any;
	selected?: boolean;
	title?: string;
}

export function getDefaultItemProps(): ItemProps {
	return cloneDeep(Object.assign(
		getDefaultBaseProps(), {
			hiddenLeftButton: false,
			hiddenRightButton: false,
			leftButton: null,
			rightButton: null,
			selected: false,
			stacked: false,
			title: '',
			widget: null
		}));
}

export const ItemComponent = (props: ItemProps) => {

	let leftButton = null;
	if (props.leftButton != null && !props.disabled) {
		leftButton = (
			<div className={`${styles.itemButton} ${(props.hiddenLeftButton) ? styles.hiddenButton : ''}`}>
				{props.leftButton != null ? props.leftButton : null}
			</div>
		);
	}

	let rightButton = null;
	if (props.rightButton != null && !props.disabled) {
		rightButton = (
			<div className={`${styles.itemButton} ${(props.hiddenRightButton) ? styles.hiddenButton : ''}`}>
				{props.rightButton != null ? props.rightButton : null}
			</div>
		);
	}

	return (
		<li
			id={props.id}
			onMouseOut={props.onMouseOut}
			onKeyDown={props.onKeyDown}
			onKeyPress={props.onKeyPress}
			className={`ui-item ${styles.item} ${props.selected ? 'ui-selected' : ''} ${props.classes.join(' ')}`}>
			{leftButton}
			<Title {...props} className={styles.itemTitle} classes={[]}>{props.title}</Title>
			{rightButton}
		</li>
	);
};
