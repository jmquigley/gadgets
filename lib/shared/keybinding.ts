/*
 * Interface defnintion used by the key map and bindings for components.  The
 * library uses the [react-hotkeys](https://www.npmjs.com/package/react-hotkeys)
 * module.  Note that all keybindings are on that component (not global), so
 * the components require focus to use the given binding.
 *
 * @module keybinding
 */

import {KeyMapOptions} from "react-hotkeys";

export interface KeyMap {
	[key: string]: string | KeyMapOptions;
}

export interface KeyHandler {
	[key: string]: (...args) => void;
}
