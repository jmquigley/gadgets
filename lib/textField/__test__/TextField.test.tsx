'use strict';

import {mount} from 'enzyme';
import * as React from 'react';
import * as sinon from 'sinon';
import {Sizing} from '../../shared';
import {getDefaultTextFieldProps, TextField} from '../index';

test('Test retrieval of TextField props object', () => {
	const props = getDefaultTextFieldProps();

	expect(props).toBeTruthy();

	expect('disabled' in props).toBe(true);
	expect(props.disabled).toBe(false);

	expect('id' in props).toBe(true);
	expect(props.id).toBe('');

	expect('sizing' in props).toBe(true);
	expect(props.sizing).toBe(Sizing.normal);

	expect('type' in props).toBe(true);
	expect(props.type).toBe('text');

	expect('usevalidation' in props).toBe(true);
	expect(props.usevalidation).toBe(false);

	expect('validators' in props).toBe(true);
	expect(props.validators instanceof Array).toBe(true);
	expect(props.validators.length).toBe(0);

	expect('visible' in props).toBe(true);
	expect(props.visible).toBe(true);
});

function validate(ctl: any) {
	expect(ctl).toBeTruthy();
	expect(ctl.prop('disabled')).toBe(false);
	expect(ctl.prop('visible')).toBe(true);
	expect(ctl.find('.ui-textfield').length).toBe(1);
	/* 	expect(ctl.find('.textFieldInput').length).toBe(1);*/
}

test('Test creation of a TextField control', () => {
	const ctl = mount(<TextField className="test-class" />);

	validate(ctl);

	expect(ctl.prop('sizing')).toBe(Sizing.normal);
	expect(ctl.prop('usevalidation')).toBe(false);
	expect(ctl.find('.test-class').length).toBe(1);
});

test('Test creation of a textfield with min/max validation', () => {
	const keypress = sinon.spy();
	const ctl = mount(
		<TextField
			maxLength="10"
			minLength="5"
			onKeyPress={keypress}
			usevalidation
			value=""
		/>
	);

	validate(ctl);

	expect(ctl.state('valid')).toBe(true);
	ctl.find('input').simulate('keyPress', {key: 'a'});
	// expect(ctl.state('valid')).toBe(false);  TODO: fix this in TextField test
	// expect(keypress.calledOnce).toBe(true);
});

// TODO: disabling the TextField control
// TODO: make the TextField invisible
// TODO: test adding/retrieving text example from the TextField control
// TODO: test validation routines
