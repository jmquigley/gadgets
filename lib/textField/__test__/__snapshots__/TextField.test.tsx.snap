// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create text, change and revert with Enter/Escape in TextField 1`] = `
<TextField
  disabled={false}
  id=""
  noborder={false}
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onValidation={[Function]}
  sizing="normal"
  style={Object {}}
  testing={true}
  tooltip=""
  type="text"
  useclear={false}
  usevalidation={false}
  validators={Array []}
  visible={true}
>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#c0c0c0",
        "color": "#000000",
        "headerBackgroundColor": "#004358",
        "headerForegroundColor": "#ffffff",
        "headerHoverColor": "#ffe11a",
        "hoverColor": "#c0c0c0",
        "inputBorderColor": "#c0c0c0",
        "tooltipBackgroundColor": "gray",
        "tooltipForegroundColor": "white",
        "transitionDelay": "0.5s",
      }
    }
  >
    <styled.div
      className="ui-textfield-container"
      id="TextField-0"
      style={Object {}}
    >
      <div
        className="ui-textfield-container sc-gqjmRU kxbhDp"
        id="TextField-0"
        style={Object {}}
      >
        <styled.div
          className="ui-textfield"
          disabled={false}
          noborder={false}
          visible={true}
        >
          <div
            className="ui-textfield sc-VigVT eMcEmF"
            disabled={false}
          >
            <styled.input
              disabled={false}
              id=""
              innerRef={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              sizing="normal"
              style={Object {}}
              testing={true}
              tooltip=""
              type="text"
              visible={true}
            >
              <input
                className="sc-gZMcBi fnEBAm"
                disabled={false}
                id=""
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                style={Object {}}
                type="text"
              />
            </styled.input>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </ThemeProvider>
</TextField>
`;

exports[`Test creating TextField and saving on blur 1`] = `
<TextField
  disabled={false}
  id=""
  noborder={false}
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onValidation={[Function]}
  sizing="normal"
  style={Object {}}
  testing={true}
  tooltip=""
  type="text"
  useclear={false}
  usevalidation={false}
  validators={Array []}
  visible={true}
>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#c0c0c0",
        "color": "#000000",
        "headerBackgroundColor": "#004358",
        "headerForegroundColor": "#ffffff",
        "headerHoverColor": "#ffe11a",
        "hoverColor": "#c0c0c0",
        "inputBorderColor": "#c0c0c0",
        "tooltipBackgroundColor": "gray",
        "tooltipForegroundColor": "white",
        "transitionDelay": "0.5s",
      }
    }
  >
    <styled.div
      className="ui-textfield-container"
      id="TextField-0"
      style={Object {}}
    >
      <div
        className="ui-textfield-container sc-gqjmRU kxbhDp"
        id="TextField-0"
        style={Object {}}
      >
        <styled.div
          className="ui-textfield"
          disabled={false}
          noborder={false}
          visible={true}
        >
          <div
            className="ui-textfield sc-VigVT eMcEmF"
            disabled={false}
          >
            <styled.input
              disabled={false}
              id=""
              innerRef={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              sizing="normal"
              style={Object {}}
              testing={true}
              tooltip=""
              type="text"
              visible={true}
            >
              <input
                className="sc-gZMcBi fnEBAm"
                disabled={false}
                id=""
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                style={Object {}}
                type="text"
              />
            </styled.input>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </ThemeProvider>
</TextField>
`;

exports[`Test creation of a TextField control 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "color": "#000000",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "tooltipBackgroundColor": "gray",
      "tooltipForegroundColor": "white",
      "transitionDelay": "0.5s",
    }
  }
>
  <styled.div
    className="ui-textfield-container"
    id="TextField-0"
    style={Object {}}
  >
    <styled.div
      className="ui-textfield test-class"
      disabled={false}
      noborder={false}
      visible={true}
    >
      <styled.input
        className="test-class"
        disabled={false}
        id=""
        innerRef={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        sizing="normal"
        style={Object {}}
        testing={true}
        tooltip=""
        type="text"
        visible={true}
      />
    </styled.div>
  </styled.div>
</ThemeProvider>
`;

exports[`Test creation of a TextField with email validation 1`] = `
<TextField
  disabled={false}
  id=""
  noborder={false}
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onValidation={[Function]}
  sizing="normal"
  style={Object {}}
  testing={true}
  tooltip=""
  type="email"
  useclear={false}
  usevalidation={true}
  validators={Array []}
  visible={true}
>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#c0c0c0",
        "color": "#000000",
        "headerBackgroundColor": "#004358",
        "headerForegroundColor": "#ffffff",
        "headerHoverColor": "#ffe11a",
        "hoverColor": "#c0c0c0",
        "inputBorderColor": "#c0c0c0",
        "tooltipBackgroundColor": "gray",
        "tooltipForegroundColor": "white",
        "transitionDelay": "0.5s",
      }
    }
  >
    <styled.div
      className="ui-textfield-container"
      id="TextField-0"
      style={Object {}}
    >
      <div
        className="ui-textfield-container sc-gqjmRU kxbhDp"
        id="TextField-0"
        style={Object {}}
      >
        <styled.div
          className="ui-textfield"
          disabled={false}
          noborder={false}
          visible={true}
        >
          <div
            className="ui-textfield sc-VigVT eMcEmF"
            disabled={false}
          >
            <styled.input
              disabled={false}
              id=""
              innerRef={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              sizing="normal"
              style={Object {}}
              testing={true}
              tooltip=""
              type="email"
              visible={true}
            >
              <input
                className="sc-gZMcBi fnEBAm"
                disabled={false}
                id=""
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                style={Object {}}
                type="email"
              />
            </styled.input>
          </div>
        </styled.div>
        <styled.div
          className="ui-textfield-validation-message"
          disabled={false}
          messageType={0}
          sizing="small"
          visible={true}
        >
          <div
            className="ui-textfield-validation-message sc-iwsKbI fzfHna"
            disabled={false}
          >
            â€‹
          </div>
        </styled.div>
      </div>
    </styled.div>
  </ThemeProvider>
</TextField>
`;

exports[`Test creation of a TextField with url validation 1`] = `
<TextField
  disabled={false}
  id=""
  noborder={false}
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onValidation={[Function]}
  sizing="normal"
  style={Object {}}
  testing={true}
  tooltip=""
  type="url"
  useclear={false}
  usevalidation={true}
  validators={Array []}
  visible={true}
>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#c0c0c0",
        "color": "#000000",
        "headerBackgroundColor": "#004358",
        "headerForegroundColor": "#ffffff",
        "headerHoverColor": "#ffe11a",
        "hoverColor": "#c0c0c0",
        "inputBorderColor": "#c0c0c0",
        "tooltipBackgroundColor": "gray",
        "tooltipForegroundColor": "white",
        "transitionDelay": "0.5s",
      }
    }
  >
    <styled.div
      className="ui-textfield-container"
      id="TextField-0"
      style={Object {}}
    >
      <div
        className="ui-textfield-container sc-gqjmRU kxbhDp"
        id="TextField-0"
        style={Object {}}
      >
        <styled.div
          className="ui-textfield"
          disabled={false}
          noborder={false}
          visible={true}
        >
          <div
            className="ui-textfield sc-VigVT eMcEmF"
            disabled={false}
          >
            <styled.input
              disabled={false}
              id=""
              innerRef={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              sizing="normal"
              style={Object {}}
              testing={true}
              tooltip=""
              type="url"
              visible={true}
            >
              <input
                className="sc-gZMcBi fnEBAm"
                disabled={false}
                id=""
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                style={Object {}}
                type="url"
              />
            </styled.input>
          </div>
        </styled.div>
        <styled.div
          className="ui-textfield-validation-message"
          disabled={false}
          messageType={0}
          sizing="small"
          visible={true}
        >
          <div
            className="ui-textfield-validation-message sc-iwsKbI fzfHna"
            disabled={false}
          >
            â€‹
          </div>
        </styled.div>
      </div>
    </styled.div>
  </ThemeProvider>
</TextField>
`;

exports[`Test creation of a textfield with min/max validation 1`] = `
<TextField
  disabled={false}
  id=""
  maxLength="10"
  minLength="5"
  noborder={false}
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onValidation={[Function]}
  sizing="normal"
  style={Object {}}
  testing={true}
  tooltip=""
  type="text"
  useclear={false}
  usevalidation={true}
  validators={Array []}
  value=""
  visible={true}
>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#c0c0c0",
        "color": "#000000",
        "headerBackgroundColor": "#004358",
        "headerForegroundColor": "#ffffff",
        "headerHoverColor": "#ffe11a",
        "hoverColor": "#c0c0c0",
        "inputBorderColor": "#c0c0c0",
        "tooltipBackgroundColor": "gray",
        "tooltipForegroundColor": "white",
        "transitionDelay": "0.5s",
      }
    }
  >
    <styled.div
      className="ui-textfield-container"
      id="TextField-0"
      style={Object {}}
    >
      <div
        className="ui-textfield-container sc-gqjmRU kxbhDp"
        id="TextField-0"
        style={Object {}}
      >
        <styled.div
          className="ui-textfield"
          disabled={false}
          noborder={false}
          visible={true}
        >
          <div
            className="ui-textfield sc-VigVT eMcEmF"
            disabled={false}
          >
            <styled.input
              disabled={false}
              id=""
              innerRef={[Function]}
              maxLength="10"
              minLength="5"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              sizing="normal"
              style={Object {}}
              testing={true}
              tooltip=""
              type="text"
              value=""
              visible={true}
            >
              <input
                className="sc-gZMcBi fnEBAm"
                disabled={false}
                id=""
                maxLength="10"
                minLength="5"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                style={Object {}}
                type="text"
                value=""
              />
            </styled.input>
          </div>
        </styled.div>
        <styled.div
          className="ui-textfield-validation-message"
          disabled={false}
          messageType={0}
          sizing="small"
          visible={true}
        >
          <div
            className="ui-textfield-validation-message sc-iwsKbI fzfHna"
            disabled={false}
          >
            â€‹
          </div>
        </styled.div>
      </div>
    </styled.div>
  </ThemeProvider>
</TextField>
`;

exports[`Test disabling of a TextField control 1`] = `
<TextField
  disabled={true}
  id=""
  noborder={false}
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onValidation={[Function]}
  sizing="normal"
  style={Object {}}
  testing={true}
  tooltip=""
  type="text"
  useclear={false}
  usevalidation={false}
  validators={Array []}
  visible={true}
>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#c0c0c0",
        "color": "#000000",
        "headerBackgroundColor": "#004358",
        "headerForegroundColor": "#ffffff",
        "headerHoverColor": "#ffe11a",
        "hoverColor": "#c0c0c0",
        "inputBorderColor": "#c0c0c0",
        "tooltipBackgroundColor": "gray",
        "tooltipForegroundColor": "white",
        "transitionDelay": "0.5s",
      }
    }
  >
    <styled.div
      className="ui-textfield-container"
      id="TextField-0"
      style={Object {}}
    >
      <div
        className="ui-textfield-container sc-gqjmRU kxbhDp"
        id="TextField-0"
        style={Object {}}
      >
        <styled.div
          className="ui-textfield"
          disabled={true}
          noborder={false}
          visible={true}
        >
          <div
            className="ui-textfield sc-VigVT gdIXPe"
            disabled={true}
          >
            <styled.input
              disabled={true}
              id=""
              innerRef={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              sizing="normal"
              style={Object {}}
              testing={true}
              tooltip=""
              type="text"
              visible={true}
            >
              <input
                className="sc-gZMcBi jCpWPK"
                disabled={true}
                id=""
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                style={Object {}}
                type="text"
              />
            </styled.input>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </ThemeProvider>
</TextField>
`;

exports[`Test making the TextField invisible 1`] = `
<TextField
  disabled={false}
  id=""
  noborder={false}
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onValidation={[Function]}
  sizing="normal"
  style={Object {}}
  testing={true}
  tooltip=""
  type="text"
  useclear={false}
  usevalidation={false}
  validators={Array []}
  visible={false}
>
  <ThemeProvider
    theme={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#c0c0c0",
        "color": "#000000",
        "headerBackgroundColor": "#004358",
        "headerForegroundColor": "#ffffff",
        "headerHoverColor": "#ffe11a",
        "hoverColor": "#c0c0c0",
        "inputBorderColor": "#c0c0c0",
        "tooltipBackgroundColor": "gray",
        "tooltipForegroundColor": "white",
        "transitionDelay": "0.5s",
      }
    }
  >
    <styled.div
      className="ui-textfield-container"
      id="TextField-0"
      style={Object {}}
    >
      <div
        className="ui-textfield-container sc-gqjmRU kxbhDp"
        id="TextField-0"
        style={Object {}}
      >
        <styled.div
          className="ui-textfield"
          disabled={false}
          noborder={false}
          visible={false}
        >
          <div
            className="ui-textfield sc-VigVT eMcEmF"
            disabled={false}
          >
            <styled.input
              disabled={false}
              id=""
              innerRef={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              sizing="normal"
              style={Object {}}
              testing={true}
              tooltip=""
              type="text"
              visible={false}
            >
              <input
                className="sc-gZMcBi bOiJhC"
                disabled={false}
                id=""
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                style={Object {}}
                type="text"
              />
            </styled.input>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </ThemeProvider>
</TextField>
`;

exports[`Test retrieval of TextField props object 1`] = `
Object {
  "disabled": false,
  "id": "",
  "noborder": false,
  "onBlur": [Function],
  "onChange": [Function],
  "onKeyDown": [Function],
  "onKeyPress": [Function],
  "onValidation": [Function],
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "tooltip": "",
  "type": "text",
  "useclear": false,
  "usevalidation": false,
  "validators": Array [],
  "visible": true,
}
`;
